### ISSUE TITLE
Master node on cluster reports having several IP addresses

### ISSUE THREAD

**Issue by llevar** [kubernetes/kubernetes#73407]
<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!-->


**What happened**:
Created a cluster following https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/ and using WeaveNet as the network. Encountered issues with getting weave to peer the nodes with the master because of the wrong IP for master being used. Issue was documented here - https://github.com/weaveworks/weave/issues/3588. After investigation we discovered that the master node reports having several IPs, which causes issues with weave as well as periodic taints because kubelet is not able to properly update status.

**What you expected to happen**:
The setup is described in detail in the linked github issue(https://github.com/weaveworks/weave/issues/3588). Additionally:

master node:
```
[root@tracker weave]# kubectl describe node tracker                                                                                                                                                                                         
Name:               tracker
Roles:              master
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/instance-type=16
                    beta.kubernetes.io/os=linux
                    failure-domain.beta.kubernetes.io/zone=AZ_1
                    kubernetes.io/hostname=tracker
                    node-role.kubernetes.io/master=
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
                    node.alpha.kubernetes.io/ttl: 0
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Sun, 27 Jan 2019 13:42:40 +0000
Taints:             node.kubernetes.io/unreachable:NoExecute
Unschedulable:      false
Conditions:
  Type                 Status    LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----                 ------    -----------------                 ------------------                ------                       -------
  NetworkUnavailable   False     Mon, 28 Jan 2019 15:30:03 +0000   Mon, 28 Jan 2019 15:30:03 +0000   WeaveIsUp                    Weave pod has set this
  MemoryPressure       False     Mon, 28 Jan 2019 15:35:20 +0000   Mon, 28 Jan 2019 15:35:20 +0000   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure         False     Mon, 28 Jan 2019 15:35:20 +0000   Mon, 28 Jan 2019 15:35:20 +0000   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure          False     Mon, 28 Jan 2019 15:35:20 +0000   Mon, 28 Jan 2019 15:35:20 +0000   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready                True      Mon, 28 Jan 2019 15:35:20 +0000   Mon, 28 Jan 2019 15:35:20 +0000   KubeletReady                 kubelet is posting ready status
  OutOfDisk            Unknown   Sun, 27 Jan 2019 13:42:40 +0000   Sun, 27 Jan 2019 13:43:57 +0000   NodeStatusNeverUpdated       Kubelet never posted node status.
Addresses:
  InternalIP:  192.168.0.15
  InternalIP:  10.35.105.10
  InternalIP:  10.35.110.13
  InternalIP:  10.35.110.13
  InternalIP:  10.35.104.5
Capacity:
 cpu:                8
 ephemeral-storage:  20960236Ki
 hugepages-1Gi:      0
 hugepages-2Mi:      0
 memory:             32781224Ki
 pods:               110
Allocatable:
 cpu:                8
 ephemeral-storage:  19316953466
 hugepages-1Gi:      0
 hugepages-2Mi:      0
 memory:             32678824Ki
 pods:               110
System Info:
 Machine ID:                 ea4bb35ac7b55a78c3c4fb8cdb4b7f65
 System UUID:                9ABFF20F-9775-44FD-8778-CE5D66C48082
 Boot ID:                    f4c58653-d452-4101-94eb-2d490da88e09
 Kernel Version:             3.10.0-693.5.2.el7.x86_64
 OS Image:                   CentOS Linux 7 (Core)
 Operating System:           linux
 Architecture:               amd64
 Container Runtime Version:  docker://18.9.1
 Kubelet Version:            v1.13.2
 Kube-Proxy Version:         v1.13.2
PodCIDR:                     10.32.0.0/24
ProviderID:                  openstack:///9abff20f-9775-44fd-8778-ce5d66c48082
Non-terminated Pods:         (10 in total)
  Namespace                  Name                                     CPU Requests  CPU Limits  Memory Requests  Memory Limits  AGE
  ---------                  ----                                     ------------  ----------  ---------------  -------------  ---
  default                    redis-master-0                           0 (0%)        0 (0%)      0 (0%)           0 (0%)         3m48s
  kube-system                calico-remove-iptables-rz67g             0 (0%)        0 (0%)      0 (0%)           0 (0%)         21h
  kube-system                etcd-tracker                             0 (0%)        0 (0%)      0 (0%)           0 (0%)         25h
  kube-system                kube-apiserver-tracker                   250m (3%)     0 (0%)      0 (0%)           0 (0%)         25h
  kube-system                kube-controller-manager-tracker          200m (2%)     0 (0%)      0 (0%)           0 (0%)         25h
  kube-system                kube-proxy-s5qxv                         0 (0%)        0 (0%)      0 (0%)           0 (0%)         25h
  kube-system                kube-scheduler-tracker                   100m (1%)     0 (0%)      0 (0%)           0 (0%)         25h
  kube-system                kubernetes-dashboard-78744cfc45-dkj8d    0 (0%)        0 (0%)      0 (0%)           0 (0%)         3m52s
  kube-system                weave-net-bgxns                          20m (0%)      0 (0%)      0 (0%)           0 (0%)         5m23s
  weave                      weave-scope-agent-ghq7p                  100m (1%)     0 (0%)      100Mi (0%)       0 (0%)         16h
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests    Limits
  --------           --------    ------
  cpu                670m (8%)   0 (0%)
  memory             100Mi (0%)  0 (0%)
  ephemeral-storage  0 (0%)      0 (0%)
Events:
  Type    Reason                   Age                     From              Message
  ----    ------                   ----                    ----              -------
  Normal  NodeHasNoDiskPressure    15m (x4833 over 6h40m)  kubelet, tracker  Node tracker status is now: NodeHasNoDiskPressure
  Normal  NodeHasSufficientMemory  10m (x4894 over 6h40m)  kubelet, tracker  Node tracker status is now: NodeHasSufficientMemory
  Normal  Starting                 6m16s                   kubelet, tracker  Starting kubelet.
  Normal  NodeAllocatableEnforced  6m15s                   kubelet, tracker  Updated Node Allocatable limit across pods
  Normal  NodeHasSufficientMemory  6m3s (x6 over 6m4s)     kubelet, tracker  Node tracker status is now: NodeHasSufficientMemory
  Normal  NodeHasNoDiskPressure    6m3s (x6 over 6m4s)     kubelet, tracker  Node tracker status is now: NodeHasNoDiskPressure
  Normal  NodeHasSufficientPID     6m3s (x6 over 6m4s)     kubelet, tracker  Node tracker status is now: NodeHasSufficientPID
  Normal  NodeReady                6m3s (x5 over 6m3s)     kubelet, tracker  Node tracker status is now: NodeReady
  Normal  Starting                 4m45s                   kubelet, tracker  Starting kubelet.
  Normal  NodeAllocatableEnforced  4m45s                   kubelet, tracker  Updated Node Allocatable limit across pods
  Normal  NodeHasSufficientMemory  4m33s (x6 over 4m33s)   kubelet, tracker  Node tracker status is now: NodeHasSufficientMemory
  Normal  NodeHasNoDiskPressure    4m33s (x6 over 4m33s)   kubelet, tracker  Node tracker status is now: NodeHasNoDiskPressure
  Normal  NodeHasSufficientPID     4m33s (x6 over 4m33s)   kubelet, tracker  Node tracker status is now: NodeHasSufficientPID
  Normal  NodeReady                4m33s (x5 over 4m33s)   kubelet, tracker  Node tracker status is now: NodeReady
```

worker node:

```
Name:               worker-2
Roles:              <none>
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/hostname=worker-2
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
                    node.alpha.kubernetes.io/ttl: 0
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Mon, 28 Jan 2019 09:49:03 +0000
Taints:             <none>
Unschedulable:      false
Conditions:
  Type                 Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----                 ------  -----------------                 ------------------                ------                       -------
  NetworkUnavailable   False   Mon, 28 Jan 2019 09:51:36 +0000   Mon, 28 Jan 2019 09:51:36 +0000   WeaveIsUp                    Weave pod has set this
  MemoryPressure       False   Mon, 28 Jan 2019 15:42:01 +0000   Mon, 28 Jan 2019 09:49:03 +0000   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure         False   Mon, 28 Jan 2019 15:42:01 +0000   Mon, 28 Jan 2019 09:49:03 +0000   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure          False   Mon, 28 Jan 2019 15:42:01 +0000   Mon, 28 Jan 2019 09:49:03 +0000   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready                True    Mon, 28 Jan 2019 15:42:01 +0000   Mon, 28 Jan 2019 09:51:54 +0000   KubeletReady                 kubelet is posting ready status
Addresses:
  InternalIP:  192.168.0.10
  Hostname:    worker-2
Capacity:
 cpu:                16
 ephemeral-storage:  20960236Ki
 hugepages-1Gi:      0
 hugepages-2Mi:      0
 memory:             65808244Ki
 pods:               110
Allocatable:
 cpu:                16
 ephemeral-storage:  19316953466
 hugepages-1Gi:      0
 hugepages-2Mi:      0
 memory:             65705844Ki
 pods:               110
System Info:
 Machine ID:                 ea4bb35ac7b55a78c3c4fb8cdb4b7f65
 System UUID:                F8A22072-94C2-4A0C-8378-C2BA112CBB57
 Boot ID:                    a18dcbc4-b64f-4757-bfb9-748874824fcc
 Kernel Version:             3.10.0-693.5.2.el7.x86_64
 OS Image:                   CentOS Linux 7 (Core)
 Operating System:           linux
 Architecture:               amd64
 Container Runtime Version:  docker://18.9.1
 Kubelet Version:            v1.13.2
 Kube-Proxy Version:         v1.13.2
PodCIDR:                     10.32.7.0/24
Non-terminated Pods:         (4 in total)
  Namespace                  Name                            CPU Requests  CPU Limits  Memory Requests  Memory Limits  AGE
  ---------                  ----                            ------------  ----------  ---------------  -------------  ---
  kube-system                calico-remove-iptables-z65ws    0 (0%)        0 (0%)      0 (0%)           0 (0%)         5h53m
  kube-system                kube-proxy-pwczg                0 (0%)        0 (0%)      0 (0%)           0 (0%)         5h53m
  kube-system                weave-net-gh2bj                 20m (0%)      0 (0%)      0 (0%)           0 (0%)         5h53m
  weave                      weave-scope-agent-pvnx5         100m (0%)     0 (0%)      100Mi (0%)       0 (0%)         5h53m
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests    Limits
  --------           --------    ------
  cpu                120m (0%)   0 (0%)
  memory             100Mi (0%)  0 (0%)
  ephemeral-storage  0 (0%)      0 (0%)
Events:              <none>
```

The master shows multiple IP addresses, while the worker shows only 1 IP address, even though the networking setup between them is exactly the same.

master

```
[root@tracker weave]# ifconfig -a
datapath: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1376
        inet6 fe80::e4c7:8cff:fe1c:5c6a  prefixlen 64  scopeid 0x20<link>
        ether e6:c7:8c:1c:5c:6a  txqueuelen 1000  (Ethernet)
        RX packets 174  bytes 8154 (7.9 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 8  bytes 648 (648.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 0.0.0.0
        inet6 fe80::42:f8ff:fe50:c69  prefixlen 64  scopeid 0x20<link>
        ether 02:42:f8:50:0c:69  txqueuelen 0  (Ethernet)
        RX packets 46803  bytes 2529295 (2.4 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 52545  bytes 420913904 (401.4 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.0.15  netmask 255.255.255.0  broadcast 192.168.0.255
        inet6 fe80::f816:3eff:fe9a:84ea  prefixlen 64  scopeid 0x20<link>
        ether fa:16:3e:9a:84:ea  txqueuelen 1000  (Ethernet)
        RX packets 5808817859  bytes 3508218276702 (3.1 TiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 5673717238  bytes 2419507091716 (2.2 TiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.35.104.5  netmask 255.255.255.0  broadcast 10.35.104.255
        inet6 fe80::f816:3eff:fe1a:52f8  prefixlen 64  scopeid 0x20<link>
        ether fa:16:3e:1a:52:f8  txqueuelen 1000  (Ethernet)
        RX packets 11846025  bytes 2207243313 (2.0 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 17552966  bytes 2141234372 (1.9 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth2: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.35.105.10  netmask 255.255.255.0  broadcast 10.35.105.255
        inet6 fe80::f816:3eff:fe2c:bd88  prefixlen 64  scopeid 0x20<link>
        ether fa:16:3e:2c:bd:88  txqueuelen 1000  (Ethernet)
        RX packets 5957873  bytes 1133760202 (1.0 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 8810778  bytes 1126103966 (1.0 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth3: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.35.110.13  netmask 255.255.255.0  broadcast 10.35.110.255
        inet6 fe80::f816:3eff:feff:6aee  prefixlen 64  scopeid 0x20<link>
        ether fa:16:3e:ff:6a:ee  txqueuelen 1000  (Ethernet)
        RX packets 11708599  bytes 1541820120 (1.4 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 11336813  bytes 159624529776 (148.6 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1  (Local Loopback)
        RX packets 688531330  bytes 149789601046 (139.5 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 688531330  bytes 149789601046 (139.5 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethd163223: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::5026:23ff:fe66:23dd  prefixlen 64  scopeid 0x20<link>
        ether 52:26:23:66:23:dd  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 13  bytes 858 (858.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethwe-bridge: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1376
        inet6 fe80::30e0:8eff:feb5:6356  prefixlen 64  scopeid 0x20<link>
        ether 32:e0:8e:b5:63:56  txqueuelen 0  (Ethernet)
        RX packets 155880  bytes 16283626 (15.5 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 146722  bytes 428090177 (408.2 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethwe-datapath: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1376
        inet6 fe80::38d8:aff:fe9f:847f  prefixlen 64  scopeid 0x20<link>
        ether 3a:d8:0a:9f:84:7f  txqueuelen 0  (Ethernet)
        RX packets 688531330  bytes 149789601046 (139.5 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 688531330  bytes 149789601046 (139.5 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethwepl56c9d35: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1376
        inet6 fe80::f8ba:11ff:febf:ba26  prefixlen 64  scopeid 0x20<link>
        ether fa:ba:11:bf:ba:26  txqueuelen 0  (Ethernet)
        RX packets 155880  bytes 16283626 (15.5 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 146722  bytes 428090177 (408.2 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethwepl9ccb0de: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1376
        inet6 fe80::88e0:cdff:feb5:6ae5  prefixlen 64  scopeid 0x20<link>
        ether 8a:e0:cd:b5:6a:e5  txqueuelen 0  (Ethernet)
        RX packets 688531330  bytes 149789601046 (139.5 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 688531330  bytes 149789601046 (139.5 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vxlan-6784: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 65470
        inet6 fe80::d437:1cff:fe0e:5314  prefixlen 64  scopeid 0x20<link>
        ether d6:37:1c:0e:53:14  txqueuelen 1000  (Ethernet)
        RX packets 7562  bytes 1375739 (1.3 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 7265  bytes 20049744 (19.1 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

weave: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1376
        inet 10.32.64.0  netmask 255.255.0.0  broadcast 10.32.255.255
        inet6 fe80::48ba:4cff:fefa:5cb3  prefixlen 64  scopeid 0x20<link>
        ether 4a:ba:4c:fa:5c:b3  txqueuelen 1000  (Ethernet)
        RX packets 127729  bytes 15528565 (14.8 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 135098  bytes 431998664 (411.9 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
[root@tracker weave]# ip route
default via 192.168.0.1 dev eth0 
10.32.0.0/16 dev weave proto kernel scope link src 10.32.64.0 
10.35.104.0/24 dev eth1 proto kernel scope link src 10.35.104.5 
10.35.105.0/24 dev eth2 proto kernel scope link src 10.35.105.10 
10.35.110.0/24 dev eth3 proto kernel scope link src 10.35.110.13 
169.254.169.254 via 192.168.0.1 dev eth0 proto static 
172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 
192.168.0.0/24 dev eth0 proto kernel scope link src 192.168.0.15 
```

node:
```
[root@worker-2 lib]# ifconfig -a                                                                                                                                                                                                            
datapath: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1376
        inet6 fe80::105f:eaff:febf:7925  prefixlen 64  scopeid 0x20<link>
        ether 12:5f:ea:bf:79:25  txqueuelen 1000  (Ethernet)
        RX packets 165  bytes 8308 (8.1 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 8  bytes 648 (648.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 0.0.0.0
        inet6 fe80::42:47ff:fe42:cdcd  prefixlen 64  scopeid 0x20<link>
        ether 02:42:47:42:cd:cd  txqueuelen 0  (Ethernet)
        RX packets 85  bytes 6461 (6.3 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 93  bytes 11478 (11.2 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

dummy0: flags=130<BROADCAST,NOARP>  mtu 1500
        ether 8a:2f:c1:67:e8:1f  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.0.10  netmask 255.255.255.0  broadcast 192.168.0.255
        inet6 fe80::f816:3eff:fe7d:e2a8  prefixlen 64  scopeid 0x20<link>
        ether fa:16:3e:7d:e2:a8  txqueuelen 1000  (Ethernet)
        RX packets 811270179  bytes 491995942236 (458.2 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 797777292  bytes 253652667279 (236.2 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.35.104.51  netmask 255.255.255.0  broadcast 10.35.104.255
        inet6 fe80::f816:3eff:fecb:62aa  prefixlen 64  scopeid 0x20<link>
        ether fa:16:3e:cb:62:aa  txqueuelen 1000  (Ethernet)
        RX packets 2242605308  bytes 23857899207239 (21.6 TiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 2070968562  bytes 161498617963 (150.4 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth2: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.35.105.45  netmask 255.255.255.0  broadcast 10.35.105.255
        inet6 fe80::f816:3eff:fe0e:960e  prefixlen 64  scopeid 0x20<link>
        ether fa:16:3e:0e:96:0e  txqueuelen 1000  (Ethernet)
        RX packets 5932561  bytes 1101317258 (1.0 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 8773364  bytes 1120235954 (1.0 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth3: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.35.110.52  netmask 255.255.255.0  broadcast 10.35.110.255
        inet6 fe80::f816:3eff:fe0e:fc21  prefixlen 64  scopeid 0x20<link>
        ether fa:16:3e:0e:fc:21  txqueuelen 1000  (Ethernet)
        RX packets 849329384  bytes 10937568374095 (9.9 TiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 561718817  bytes 7587353455856 (6.9 TiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1  (Local Loopback)
        RX packets 46177096  bytes 76069143119 (70.8 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 46177096  bytes 76069143119 (70.8 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethwe-bridge: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1376
        inet6 fe80::dc7a:5eff:fe1a:a0b9  prefixlen 64  scopeid 0x20<link>
        ether de:7a:5e:1a:a0:b9  txqueuelen 0  (Ethernet)
        RX packets 69775  bytes 9006766 (8.5 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 78142  bytes 225947896 (215.4 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethwe-datapath: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1376
        inet6 fe80::18a7:64ff:fe69:e513  prefixlen 64  scopeid 0x20<link>
        ether 1a:a7:64:69:e5:13  txqueuelen 0  (Ethernet)
        RX packets 46177096  bytes 76069143119 (70.8 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 46177096  bytes 76069143119 (70.8 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vxlan-6784: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 65470
        inet6 fe80::a4aa:65ff:fe11:e7b6  prefixlen 64  scopeid 0x20<link>
        ether a6:aa:65:11:e7:b6  txqueuelen 1000  (Ethernet)
        RX packets 79318  bytes 20439674 (19.4 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 87242  bytes 237228892 (226.2 MiB)
        TX errors 0  dropped 7 overruns 0  carrier 0  collisions 0

weave: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1376
        inet 10.32.160.0  netmask 255.255.0.0  broadcast 10.32.255.255
        inet6 fe80::ec4f:3aff:fe09:e0ea  prefixlen 64  scopeid 0x20<link>
        ether ee:4f:3a:09:e0:ea  txqueuelen 1000  (Ethernet)
        RX packets 69747  bytes 8027832 (7.6 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 78134  bytes 225947248 (215.4 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

[root@worker-2 lib]# ip route
default via 192.168.0.1 dev eth0 
10.32.0.0/16 dev weave proto kernel scope link src 10.32.160.0 
10.35.104.0/24 dev eth1 proto kernel scope link src 10.35.104.51 
10.35.105.0/24 dev eth2 proto kernel scope link src 10.35.105.45 
10.35.110.0/24 dev eth3 proto kernel scope link src 10.35.110.52 
169.254.169.254 via 192.168.0.1 dev eth0 proto static 
172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 
192.168.0.0/24 dev eth0 proto kernel scope link src 192.168.0.10 
```

kubelet logs on master have this:

```
Jan 28 14:55:43 tracker kubelet[7634]: E0128 14:55:43.497126    7634 kubelet_node_status.go:380] Error updating node status, will retry: failed to patch status "{\"status\":{\"$setElementOrder/addresses\":[{\"type\":\"InternalIP\"},{\"type\":\"InternalIP\"},{\"type\":\"InternalIP\"},{\"type\":\"InternalIP\"}],\"$setElementOrder/conditions\":[{\"type\":\"NetworkUnavailable\"},{\"type\":\"MemoryPressure\"},{\"type\":\"DiskPressure\"},{\"type\":\"PIDPressure\"},{\"type\":\"Ready\"},{\"type\":\"OutOfDisk\"}],\"addresses\":[{\"address\":\"10.35.105.10\",\"type\":\"InternalIP\"},{\"address\":\"192.168.0.15\",\"type\":\"InternalIP\"},{\"address\":\"10.35.104.5\",\"type\":\"InternalIP\"},{\"address\":\"10.35.110.13\",\"type\":\"InternalIP\"}],\"conditions\":[{\"lastHeartbeatTime\":\"2019-01-28T14:55:43Z\",\"lastTransitionTime\":\"2019-01-28T14:55:43Z\",\"message\":\"kubelet has sufficient memory available\",\"reason\":\"KubeletHasSufficientMemory\",\"status\":\"False\",\"type\":\"MemoryPressure\"},{\"lastHeartbeatTime\":\"2019-01-28T14:55:43Z\",\"lastTransitionTime\":\"2019-01-28T14:55:43Z\",\"message\":\"kubelet has no disk pressure\",\"reason\":\"KubeletHasNoDiskPressure\",\"status\":\"False\",\"type\":\"DiskPressure\"},{\"lastHeartbeatTime\":\"2019-01-28T14:55:43Z\",\"lastTransitionTime\":\"2019-01-28T14:55:43Z\",\"message\":\"kubelet has sufficient PID available\",\"reason\":\"KubeletHasSufficientPID\",\"status\":\"False\",\"type\":\"PIDPressure\"},{\"lastHeartbeatTime\":\"2019-01-28T14:55:43Z\",\"lastTransitionTime\":\"2019-01-28T14:55:43Z\",\"message\":\"kubelet is posting ready status\",\"reason\":\"KubeletReady\",\"status\":\"True\",\"type\":\"Ready\"}]}}" for node "tracker": Node "tracker" is invalid: status.addresses[2]: Duplicate value: core.NodeAddress{Type:"InternalIP", Address:"10.35.110.13"}
```

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
[root@tracker ~]# kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.2", GitCommit:"cff46ab41ff0bb44d8584413b598ad8360ec1def", GitTreeState:"clean", BuildDate:"2019-01-10T23:35:51Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T20:56:12Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
Openstack
- OS (e.g. from /etc/os-release):
```
[root@tracker ~]# cat /etc/os-release 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
- Kernel (e.g. `uname -a`):
```
[root@tracker ~]# uname -a
Linux tracker 3.10.0-693.5.2.el7.x86_64 #1 SMP Fri Oct 20 20:32:50 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools:
- Others:

**Comment by llevar** [comment]
@kubernetes/sig-network-bugs /sig network

**Comment by brtknr** [comment]
I can confirm that I have the same problem but with flannel backend. The nodes keep getting allocated different IPs. I have gotten around the problem by prioritising DNS as the way to address nodes instead of InternalIP.

**Comment by caseydavenport** [comment]
> Jan 28 14:55:43 tracker kubelet[7634]: E0128 14:55:43.497126    7634 kubelet_node_status.go:380] Error updating node status, will retry

Is this error repeated or does it go away after the retry?

It does look like that address appears twice in the master output, which feels wrong to me, so there might be a bug here. But, I think those addresses are likely coming from the Openstack cloud provider.

My understanding is that Kubernetes will pick node IP addresses based on one of a few things:
- Cloud provider, if configured on that node
- the `--node-ip` argument if passed to the kubelet
- the first non-loopback IPv4 address it finds.

It's possible the nodes are getting different IPs because they are configured differently from the master - do they all have the openstack cloud provider enabled?

I don't think it's inherently a bug that Kubernetes has multiple IP addresses - if the node has multiple IP addresses then I think that is expected. However, it might be a bug in the openstack cloud provider if it is reporting the same address multiple times. Maybe someone from @kubernetes/sig-openstack-bugs can look into that part?

I'm not actually sure what the expectation is for clients when picking an address when there are multiple `InternalIPs`

**Comment by caseydavenport** [comment]
Given my current understanding, I don't think this is a sig-network bug - if anything, there is a bug here in the cloud provider for trying to set duplicate/wrong IP addresses.

/remove-sig network

**Comment by djgilcrease** [comment]
I think this may have been fixed by https://github.com/kubernetes/kubernetes/pull/70805 which is included in 1.14.0 at least it was for my deployments
